### PR TITLE
Log API errors

### PR DESCRIPTION
**Pull request by shuplenkov**
### Issue being fixed or implemented
- Log API errors

### What was done
- wrapToErrorHandler now logs errors throuth the Logger

### CODE REVIEW

File: lib/api/jsonRpc/wrapToErrorHandlerFactory.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    async function apiMethodErrorHandler(params) {
+      try {
+        return await method(params);
+      } catch (e) {
+        if (e instanceof InvalidParamsError) {
+          throw createError(errors.INVALID_PARAMS, e.message, e.data);
+        }
+
+        logger.error(`Error in ${method.name} API method`, { params }, e);
```
Comment (shumkov) [inline lib/api/jsonRpc/wrapToErrorHandlerFactory.js]: I would pass a method name to data as well.

File: lib/api/jsonRpc/wrapToErrorHandlerFactory.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    async function apiMethodErrorHandler(params) {
+      try {
+        return await method(params);
+      } catch (e) {
+        if (e instanceof InvalidParamsError) {
+          throw createError(errors.INVALID_PARAMS, e.message, e.data);
+        }
+
+        const data = Object.assign({ method: method.name }, { params });
```
Comment (shumkov) [inline lib/api/jsonRpc/wrapToErrorHandlerFactory.js]: Why not just:
```js
logger.error(`Error in ${method.name} API method`, { method: method.name, params }, e);
```

Comment (shuplenkov) [reply lib/api/jsonRpc/wrapToErrorHandlerFactory.js]: yeah, looks weird...

File: lib/api/jsonRpc/wrapToErrorHandlerFactory.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+const { Server: { errors } } = require('jayson');
+const createError = require('./createError');
+const InvalidParamsError = require('../InvalidParamsError');
+
+/**
+ *
+ * @param logger
```
Comment (shumkov) [inline lib/api/jsonRpc/wrapToErrorHandlerFactory.js]:
```
@param {Logger} logger
```
?

**Pull request merged**